### PR TITLE
Add sentence to textprops tutorial mentioning mathtext rcParams

### PR DESCRIPTION
**Pull request by geib7592**
## PR Summary

Addresses issue https://github.com/matplotlib/matplotlib/issues/8702.

Minor change to the text_props tutorial to point out that the math font needs to be set with the `mathtext` rcParams. I also added a label to the math tutorial so I could link to it.

I'm a new contributor, and this is my first PR. Please let me know if I've done anything incorrectly, and thanks for your patience! If all goes well, I'd like to get more involved! 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
Thanks for the PR!  :+1: to the content, but sphinx is not happy:

```
Warning, treated as error:
/home/travis/build/matplotlib/matplotlib/doc/tutorials/text/text_props.rst:162:undefined label: mathtext_fonts (if the link has no caption the label must precede a section header)
```

and I am not sure why....

**Comment by tacaswell** [comment]
I wonder if this is an order of compilation issue with sphinx gallery? attn @choldgraf ?

**Comment by choldgraf** [comment]
huh....that's weird. It looks like the label is defined properly. Do the generated RST files all look correct? (inside of `tutorials`?

**Comment by geib7592** [comment]
Oh yeah, it looks likes the generated mathtext RST is putting a bunch of stuff in a code block that shouldn't be (including the definition of the label I'm trying to reference.)

**Comment by choldgraf** [comment]
maybe it's an indentation problem then - double check that everything is indented properly!

**Comment by geib7592** [comment]
Comparing `mathtext.py` and `text_props.py`, I noticed that `mathtext.py` didn't start the comment section with a line of ####. I added that, and it compiled further. Then there was a missing # on line 36. Once those were in, the documentation compiled.

Also, I had to remove some unused footnotes in a in order to get the documentation to compile. Not sure if you guys got that error too. I'm running Sphinx 1.6.2.

**Comment by tacaswell** [comment]
I think you are working from and old checkout, those references were already removed in 3649684037a827476397d4606df866eecf7ed45d

**Comment by tacaswell** [comment]
This is a candidate for squash-merging.

@geib7592 If you are comfortable doing so, it might be worth rebasing this PR.

Thanks for working through the sphinx issues :smile:

**Comment by geib7592** [comment]
Darnit. Well, I'm learning as I go. So I attempted to rebase, and now this PR contains commits that are unrelated to what I changed. Is this fixable, or should I just start over?

**Comment by tacaswell** [comment]
```
git branch backup  # create a branch where you are just to be safe!
git rebase -i  origin/master  # follow the directions to drop the commits you do not want
git push --force your_fork
git branch -d backup  # if you are happy
```

**Comment by tacaswell** [comment]
And creating git-snarls is something that happens occasionally to everyone (I am currently at scipy watching some tutorial presenters sort out some git issues :sunglasses: )

**Comment by choldgraf** [comment]
@tacaswell is there any resource we can point contributors to that explains more clearly how to rebase? If not I can write something up...rebasing is pretty confusing the first several times it's done...

**Comment by tacaswell** [comment]
@choldgraf see https://github.com/matplotlib/matplotlib/issues/8398 We had a nice write up (to be fair I am biased because I wrote it) that some how got deleted.

**Comment by choldgraf** [comment]
ah cool, that is v useful!

I had a quick comment on that text but I'll add it in the issue you opened.

**Comment by tacaswell** [comment]
@geib7592 I think you missed the force push step (and did a `git merge origin branch` because the CLI suggested you do that) ;)

**Comment by geib7592** [comment]
Thanks, it looked like I was digging myself deeper into the abyss. But I think it looks okay now.

**Comment by choldgraf** [comment]
Little known fact, when Nietzsche said

> if you stare into the abyss, the abyss stares back at you

he was actually talking about rebasing off of a stale git branch

File: tutorials/text/text_props.py [hunk 1]
```diff
@@ -143,7 +143,9 @@
 #  Default Font
 # ==============
 #
-# The base default font is controlled by a set of rcParams:
+# The base default font is controlled by a set of rcParams. To set the font
+# for mathematical expressions, use the rcParams begining with ``mathtext``
```
Comment (QuLogic) [inline tutorials/text/text_props.py]: beginning

**Comment by tacaswell** [comment]
:+1:  You are now through the worst of the git learning curve.  http://tom.preston-werner.com/2009/05/19/the-git-parable.html is my favorite reference for this sort of thing.

**Pull request merged**